### PR TITLE
Check localhost is resolved to ipv4 or ipv6 and update test accordingly.

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

**Comment by yang-g** [comment]
It turns out some tests such as xds_e2e_test have changed from the internal version and thus I re-applied the changes manually again. This should work now. PTAL Thanks. @markdroth

File: test/core/util/resolve_localhost_ip46.cc [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (markdroth) [inline test/core/util/resolve_localhost_ip46.cc:1]: Nit: Please use C++-style comments.

File: test/core/util/resolve_localhost_ip46.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (markdroth) [inline test/core/util/resolve_localhost_ip46.h:1]: Same here.

File: test/cpp/end2end/xds_end2end_test.cc [hunk 3]
```diff
@@ -2041,6 +2024,32 @@ class XdsEnd2endTest : public ::testing::TestWithParam<TestType> {
...
+  default_cluster_.set_type(Cluster::EDS);
+  auto* eds_config = default_cluster_.mutable_eds_cluster_config();
+  eds_config->mutable_eds_config()->mutable_ads();
+  eds_config->set_service_name(kDefaultEdsServiceName);
+  default_cluster_.set_lb_policy(Cluster::ROUND_ROBIN);
+  if (enable_load_reporting) {
+    default_cluster_.mutable_lrs_server()->mutable_self();
+  }
+  SetCdsResource(default_cluster_);
+}
```
Comment (markdroth) [inline test/cpp/end2end/xds_end2end_test.cc:2052]: Please add a blank line after this one.

File: test/cpp/end2end/xds_end2end_test.cc [hunk 4]
```diff
@@ -2041,6 +2024,32 @@ class XdsEnd2endTest : public ::testing::TestWithParam<TestType> {
   grpc_channel_args xds_channel_args_;
 };
 
+AdsServiceImpl::AdsServiceImpl(bool enable_load_reporting)
```
Comment (markdroth) [inline test/cpp/end2end/xds_end2end_test.cc:2027]: Please add a TODO assigned to me to move all of the code that deals with default resource contents out of `AdsServiceImpl` and into `XdsEnd2EndTest`.

**Review by markdroth — APPROVED** [review]
This looks great!  Feel free to merge after addressing the comments.

**Pull request merged**